crates/bitwarden-core/src/key_management/master_password.rs

```diff
@@ -0,0 +1,380 @@
+//! Mobile specific master password operations
+//!
+//! This module contains the data structures and error handling for master password unlock
+//! operations.
+
+use std::num::NonZeroU32;
+
+use bitwarden_api_api::models::{
+    master_password_unlock_response_model::MasterPasswordUnlockResponseModel, KdfType,
+};
+use bitwarden_crypto::{CryptoError, EncString, Kdf};
+use bitwarden_error::bitwarden_error;
+use serde::{Deserialize, Serialize};
+#[cfg(feature = "wasm")]
+use wasm_bindgen::prelude::*;
+
+use crate::{require, MissingFieldError};
+
+/// Error for master password related operations.
+#[allow(missing_docs)]
+#[bitwarden_error(flat)]
+#[derive(Debug, thiserror::Error)]
+pub enum MasterPasswordError {
+    #[error(transparent)]
+    Crypto(#[from] CryptoError),
+    #[error(transparent)]
+    MissingField(#[from] MissingFieldError),
+}
+
+/// Represents the data required to unlock with the master password.
+#[allow(missing_docs)]
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
+#[cfg_attr(
+    feature = "wasm",
+    derive(tsify::Tsify),
+    tsify(into_wasm_abi, from_wasm_abi)
+)]
+pub struct MasterPasswordUnlockData {
+    pub kdf: Kdf,
+    pub master_key_wrapped_user_key: EncString,
+    pub salt: String,
+}
+
+impl TryFrom<MasterPasswordUnlockResponseModel> for MasterPasswordUnlockData {
+    type Error = MasterPasswordError;
+
+    fn try_from(response: MasterPasswordUnlockResponseModel) -> Result<Self, Self::Error> {
+        let kdf = match response.kdf.kdf_type {
```

Comment:
If you do apply the suggestion from above, then this would require mapping to the different error types